frontend/src/js/components/common/ConfirmModal/ConfirmModal.tsx

```diff
@@ -0,0 +1,77 @@
+// Copyright 2024 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+import { FormEvent, useState } from 'react';
+
+import { Button, Dialog, DialogActions, DialogContent, DialogContentText, DialogTitle, TextField } from '@mui/material';
+
+interface ConfirmModalProps {
+  close: () => void;
+  onConfirm: () => void;
+  toType: string;
+  header: string;
+  description: string;
+  open: boolean;
+  className?: string;
+  width?: number;
+}
+export const ConfirmModal = (props: ConfirmModalProps) => {
+  const { close, onConfirm, className, toType, header, description, open, width } = props;
+  const [inputValue, setInputValue] = useState<string>('');
+  const color = { color: '#000' };
+  return (
+    <Dialog
+      className={className || ''}
+      open={open}
+      onClose={close}
+      sx={{
+        '& .MuiDialog-container': {
+          '& .MuiPaper-root': {
+            width: '100%',
+            maxWidth: `${width}px` // Set your width here
+          }
+        }
+      }}
+      PaperProps={{
+        component: 'form',
+        onSubmit: (event: FormEvent<HTMLFormElement>) => {
+          event.preventDefault();
+          onConfirm();
+          close();
+        }
+      }}
+    >
+      <DialogTitle sx={color}>{header}</DialogTitle>
+      <DialogContent>
+        <DialogContentText sx={color}>{description}</DialogContentText>
+        <DialogContentText sx={color}>Type &#39;{toType}&#39; below to continue</DialogContentText>
+        <TextField
+          value={inputValue}
+          onChange={e => setInputValue(e.target.value)}
+          autoFocus
+          required
+          name="text"
```

Comment:
Maybe
```suggestion
          name="confirmation-text"
```
or something else more descriptive?